Cannot split segment - viomi

<!-- Text between these tags doesn't get displayed so you don't have to remove it -->
## Describe the bug
<!--
A clear and concise description of what the bug is.
-->
I'm unable to separate segments within the segmentation editor. 
I was able to do it initially when it was introduced, but not what I was trying to edit my room layout, it didn't let me do it.
I even tried joining the living room with the kitchen, but now I'm even unable to split those :) 
## To Reproduce
<!--
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
-->

### Screenshots
<!--If applicable, add screenshots to help explain your problem.-->
<img width="1692" alt="image" src="https://user-images.githubusercontent.com/46720448/113755116-3aff1a80-9710-11eb-839d-e27885ea0471.png">

### Vacuum Model
<!-- e.g. Roborock S5/Dreame D9/etc -->
viomi v7
### Valetudo Version
<!-- e.g. 2021.1.1 -->
77e712813191bc427306c80bf79f1b81f47202fd
## Expected behavior
<!--
A clear and concise description of what you expected to happen.
-->
Ability to split segments. 
## Additional context
<!-- Add any other context about the problem here. -->
```
 Apr  6 17:34:34 2021 daemon.err valetudo[697]: [2021-04-06T15:34:34.637Z] [ERROR] Failed to poll Attributes MiioTimeoutError: request timed out:{"method":"get_consumables","params":[],"id":148}
Tue Apr  6 17:34:34 2021 daemon.err valetudo[697]:     at Timeout.timeout [as _onTimeout] (/snapshot/Valetudo/lib/miio/MiioSocket.js:197:32)
Tue Apr  6 17:34:34 2021 daemon.err valetudo[697]:     at listOnTimeout (internal/timers.js:554:17)
Tue Apr  6 17:34:34 2021 daemon.err valetudo[697]:     at processTimers (internal/timers.js:497:7)
Tue Apr  6 17:34:35 2021 daemon.err valetudo[697]: [2021-04-06T15:34:35.675Z] [WARN] Error while splitting segment {
Tue Apr  6 17:34:35 2021 daemon.err valetudo[697]:   body: {
Tue Apr  6 17:34:35 2021 daemon.err valetudo[697]:     action: 'split_segment',
Tue Apr  6 17:34:35 2021 daemon.err valetudo[697]:     pA: { x: 1870, y: 1530 },
Tue Apr  6 17:34:35 2021 daemon.err valetudo[697]:     pB: { x: 2045, y: 1530 },
Tue Apr  6 17:34:35 2021 daemon.err valetudo[697]:     segment_id: 12
Tue Apr  6 17:34:35 2021 daemon.err valetudo[697]:   },
Tue Apr  6 17:34:35 2021 daemon.err valetudo[697]:   e: MiioTimeoutError: request timed out:{"method":"arrange_room","params":{"lang":"en","pointArr":[[1,"-1.3_4.7","0.45_4.7"]],"roomId":12,"type":1},"id":147}
Tue Apr  6 17:34:35 2021 daemon.err valetudo[697]:       at Timeout.timeout [as _onTimeout] (/snapshot/Valetudo/lib/miio/MiioSocket.js:197:32)
Tue Apr  6 17:34:35 2021 daemon.err valetudo[697]:       at listOnTimeout (internal/timers.js:554:17)
Tue Apr  6 17:34:35 2021 daemon.err valetudo[697]:       at processTimers (internal/timers.js:497:7)
Tue Apr  6 17:34:35 2021 daemon.err valetudo[697]: }
```

Segment IDs I have : `10, 11, 12`

--- a/lib/robots/viomi/capabilities/ViomiMapSegmentEditCapability.js
+++ b/lib/robots/viomi/capabilities/ViomiMapSegmentEditCapability.js
@@ -47,16 +47,21 @@ class ViomiMapSegmentEditCapability extends MapSegmentEditCapability {
             throw new Error("Can't join segments because the map was not parsed yet");
         }
 
-        await this.robot.sendCommand("arrange_room", {
-            lang: this.lang,
-            mapId: this.robot.state.map.metaData.vendorMapId,
-            roomArr: [[segmentA.id, segmentB.id]],
-            type: this.mapActions.JOIN_SEGMENT_TYPE
-        }, {
-            timeout: 5000
-        }).finally(() => {
+        try {
+            const result = await this.robot.sendCommand("arrange_room", {
+                lang: this.lang,
+                mapId: this.robot.state.map.metaData.vendorMapId,
+                roomArr: [[segmentA.id, segmentB.id]],
+                type: this.mapActions.JOIN_SEGMENT_TYPE
+            }, {
+                timeout: 5000
+            });
+            if (result === ["fail"]) {
+                throw new Error("Segments must be adjacent!");
+            }
+        } finally {
             this.robot.pollMap();
-        });
+        }
     }
 
     /**
@@ -74,21 +79,26 @@ class ViomiMapSegmentEditCapability extends MapSegmentEditCapability {
             throw new Error("Can't split segment because the map was not parsed yet");
         }
 
-        await this.robot.sendCommand("arrange_room", {
-            lang: this.lang,
-            mapId: this.robot.state.map.metaData.vendorMapId,
-            pointArr: [[
-                1,
-                this.pointToViomiString(ViomiMapParser.positionToViomi(pA.x, pA.y)),
-                this.pointToViomiString(ViomiMapParser.positionToViomi(pB.x, pB.y))
-            ]],
-            roomId: segment.id,
-            type: this.mapActions.SPLIT_SEGMENT_TYPE
-        }, {
-            timeout: 5000
-        }).finally(() => {
+        try {
+            const result = await this.robot.sendCommand("arrange_room", {
+                lang: this.lang,
+                mapId: this.robot.state.map.metaData.vendorMapId,
+                pointArr: [[
+                    1,
+                    this.pointToViomiString(ViomiMapParser.positionToViomi(pA.x, pA.y)),
+                    this.pointToViomiString(ViomiMapParser.positionToViomi(pB.x, pB.y))
+                ]],
+                roomId: segment.id,
+                type: this.mapActions.SPLIT_SEGMENT_TYPE
+            }, {
+                timeout: 5000
+            });
+            if (result === ["fail"]) {
+                throw new Error("Split segment is too small!");
+            }
+        } finally {
             this.robot.pollMap();
-        });
+        }
     }
 }
 

--- a/lib/robots/viomi/capabilities/ViomiMapSegmentEditCapability.js
+++ b/lib/robots/viomi/capabilities/ViomiMapSegmentEditCapability.js
@@ -76,7 +76,7 @@ class ViomiMapSegmentEditCapability extends MapSegmentEditCapability {
 
         await this.robot.sendCommand("arrange_room", {
             lang: this.lang,
-            mapId: this.robot.state.map.metaData.id,
+            mapId: this.robot.state.map.metaData.vendorMapId,
             pointArr: [[
                 1,
                 this.pointToViomiString(ViomiMapParser.positionToViomi(pA.x, pA.y)),

